Update components' styles and markup

feelthatvib3 opened the pull request:
## Description

Improved the stylization of the page as well as the markup of the components.

## Related Issues

#324

## Changes Proposed

To begin, modifications were applied to the `App.js` markup to center and restrict the width of the primary content on the page, resulting in a cleaner appearance with added whitespace.

Subsequently, enhancements were made to the `Search.jsx` component in response to the reported issue #324. These adjustments resulted in changes to the overall page flow to ensure proper placement of the input. Firstly, CSS properties like `margin-left/right/top: **rem` were eliminated from `Search.css`, `Pagination.css`, etc., which previously imposed an unnatural flow on elements with arbitrary values. Notably, the search icon was removed, as the search functionality activates immediately upon user input, making the icon unnecessary. The placeholder text was also modified to simply read `Search...` (formerly `Search user by name, location, or skills`) to accommodate smaller screens and prevent truncation. Furthermore, a new information block (`<div className="info">`) was introduced below the input to provide instructions on how to use the search functionality.

Another enhancement that was introduced to the `Sidebar.jsx` component is applying the CSS property `position: sticky`. This change ensures that the sidebar retains its width and remains visible, including the logo and, significantly, the prominent call-to-action (CTA) button labeled `Add your profile`. Consequently, users can easily access the button and add their profiles irrespective of how far they have scrolled down the page, leading to an enhanced user experience (UX).

Additionally, enhancements were applied to the content's markup and styles within the `Sidebar.jsx` component by getting rid of the redundant code and optimizing class names and styles.

## Checklist

- [x] I have read and followed the [Contribution Guidelines](https://github.com/shyamtawli/devFind/blob/master/CONTRIBUTING.md).
- [x] All new and existing tests passed.
- [x] I have updated the documentation to reflect the changes I've made.
- [x] My code follows the code style of this project.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots 

Main page BEFORE: 
![image](https://github.com/shyamtawli/devFind/assets/88098862/ab6243f1-fae0-4ffa-861a-67c23c79ed55) 

Main page AFTER:
![image](https://github.com/shyamtawli/devFind/assets/88098862/a0bb3b0e-581a-42a2-b7e7-2ce610a2f83b)

shyamtawli commented:
Thanks @feelthatvib3 for the contribution. I will review it and let you know, if any changes needed. Give me some time.

shyamtawli commented:
@feelthatvib3 
1. In my screen the alignment is not working properly. 
2. Sidebar title and logo should be align with search-bar.

or you can first fix the search-bar issue first than we can discuss about other things? You can open a PR for just the search-bar problem.

![devf](https://github.com/shyamtawli/devFind/assets/104145978/346559da-d169-4278-b7fc-b89c57e5730a)

feelthatvib3 commented:
> @feelthatvib3
> 
> 1. In my screen the alignment is not working properly.
> 2. Sidebar title and logo should be align with search-bar.
> 
> or you can first fix the search-bar issue first than we can discuss about other things? You can open a PR for just the search-bar problem.
> 
> ![devf](https://user-images.githubusercontent.com/104145978/256830775-346559da-d169-4278-b7fc-b89c57e5730a.PNG)

A quick update. I managed to fix the alignment of sidebar to the search bar. The problem with the profile cards alignment basically lays in the `Profile.css` on line 23: `.profile-card .profile-photo { width: 6.1rem }`. Removing this property partially fixes the issue, however the card is still too large in width to fit within the screen. By the way, changing the card UI was a part of my goal here, so maybe we could make things work this way.

What I suggest we do now:

1. I add an alignment fix of the sidebar to the search bar.
2. I remove the above-mentioned property on the line 23 in `Profile.css` to smooth things out a little bit.
3. In the meantime, I'll come up with an idea of redesigning the card UI.

shyamtawli commented:
The alignment is still the same and sidebar is not looking good.

what is am thinking that we can proceed one by one.
First you can fix the searchbar issue that is removing the icon, improve placeholder and add the line you have added below the search bar. Afterwards we can discuss what more enhancement can be done. 

I appreciate your changes and ideas.

feelthatvib3 commented:
> The alignment is still the same and sidebar is not looking good.
> 
> what is am thinking that we can proceed one by one. First you can fix the searchbar issue that is removing the icon, improve placeholder and add the line you have added below the search bar. Afterwards we can discuss what more enhancement can be done.
> 
> I appreciate your changes and ideas.

Fixed the alignment in the [commit](https://github.com/shyamtawli/devFind/pull/326/commits/f6af67b5a3f4110c1b5730e0a52d05d85ae1b8ae) above. Search bar is also fixed and the new line below it has also been added. Previous commits that have been made also relate to the search component positioning fix as well as some minor fixes in the code. Basically, I believe this is everything we can do in terms of this PR.

feelthatvib3 commented:
> @feelthatvib3 All Good but there is a problem. Try searching a profile. You will notice sidebar width is increasing.

I seems to me like the issue is not related to the sidebar, but to the cards themselves. I guess I open a new issue regarding this problem then?

shyamtawli commented:
Can you connect me on any social media?

feelthatvib3 commented:
> Can you connect me on any social media?

Yeah, does telegram sound good for you?
